kms/keys.go

```diff
@@ -0,0 +1,236 @@
+package kms
+
+import (
+	"go.uber.org/zap"
+	"google.golang.org/api/cloudkms/v1"
+)
+
+type CryptoKey struct {
+	ID               uint `gorm:"primarykey"`
+	ProjectID        string
+	CreateTime       string
+	Name             string
+	NextRotationTime string
+
+	PrimaryAlgorithm string
+
+	CaviumCerts          []*CryptoKeyCertificateChainsCaviumCerts          `gorm:"constraint:OnDelete:CASCADE;foreignKey:ID"`
```

Comment:
no need for the `foreignKey:ID` `ID` is the primary key and if you name the foreignkey `CryptoKeyID` it is detected automatically